### PR TITLE
make the `ping` detection filterable

### LINKED ISSUE #224: Support webmention type
Webmentions aren't in WordPress Core, and exist only as a plugin. Spam webmention are not a thing yet, hopefully never. 

Webmentions are a W3C Specification and a modern successor to pingbacks. 

The only real thing Antispam Bee needs to do is be aware of the webmention comment type.

### PR DESCRIPTION
**Pull request by pfefferle**
For the Webmention plugin, we introduce multiple new comment types, so we need a way to hook into the `is_ping` check to return `true` on a Webmention ping.

fixes #224

### CODE REVIEW

**Comment by Zodiac1978** [comment]
Any reason why you filter the end result and not the types array?

**Comment by pfefferle** [comment]
Sure. A filter to the comment type, would force new ping services to use one of the existing or a new comment type and I see this very limiting. Webmention, for example, is only the protocol and it supports several comment-types, including `comment`. If we only provide a filter for the comment types, I have to add `comment`, so the (security) mechanism to also check the request path (https://github.com/pluginkollektiv/antispam-bee/pull/381/files#diff-aed4fab3c070d2e8344bc39a125a8ddbf7ba1bda22124e940f5d9afe63e62732R1126) would not work any more or otherwise it is not possible to add `comment` webmentions via endpoint.

With the result filter, developers can also check other comment metas including the comment-type.

**Comment by pfefferle** [comment]
Should I additionally add a filter for the comment types?

**Comment by Zodiac1978** [comment]
> Should I additionally add a filter for the comment types?

Maybe.

I was just confused by the description:

> For the Webmention plugin, we introduce multiple new comment types

I don't know more about the logic behind Webmention and thought we just need to add more types to the array, so I thought it would be easier to just add the types to the array. If there is a Webmention type with comment which would collide with our logic it seems absolute okay to go this way.

For other uses cases it could be useful to add more comment types, but we could add this if this is an actual feature request.

**Comment by florianbrinkmann** [comment]
> Any reason why you filter the end result and not the types array?

I had the same question in mind when looking at the code :D

> Sure. A filter to the comment type, would force new ping services to use one of the existing or a new comment type and I see this very limiting. Webmention, for example, is only the protocol and it supports several comment-types, including comment. If we only provide a filter for the comment types, I have to add comment, so the (security) mechanism to also check the request path (https://github.com/pluginkollektiv/antispam-bee/pull/381/files#diff-aed4fab3c070d2e8344bc39a125a8ddbf7ba1bda22124e940f5d9afe63e62732R1126) would not work any more or otherwise it is not possible to add comment webmentions via endpoint.

In my understanding, it should not make a difference whether there is one filter that filters the result, like now, or there is one filter for the types array, or am I missing something? :thinking: Plugins that use other types would have to use the filter regardless of which version is added.

**Review by Zodiac1978 — APPROVED** [review]
LGTM, but I recommend a second pair of eyes. (pandemic tired mode)

**Comment by florianbrinkmann** [comment]
> For other uses cases it could be useful to add more comment types, but we could add this if this is an actual feature request.

Sounds like a good approacth, @pfefferle you know best what is needed for the webmention use case, so LGTM, too :+1:

**Review by florianbrinkmann — APPROVED** [review]

**Comment by florianbrinkmann** [comment]
Because I would likely forget it: Can we already define a value for `@since`, or is that something we need to remember before the next release?

**Comment by pfefferle** [comment]
@Zodiac1978 @florianbrinkmann I would also like to add the `@since`. Is there already a plan if the next release will be a major, minor or a patch?

**Comment by florianbrinkmann** [comment]
I guess that a 2.10 will likely be the next release. At least I am not aware of bugs that would need a patch release. Maybe we should think about a check that we could add to the release action to search for `@since <TBD>` (or similar) in the codebase before a release is created, so we do not need to always define that when merging (or remembering it) :)

**Comment by Zodiac1978** [comment]
> Is there already a plan if the next release will be a major, minor or a patch?

No, but in retrospective the last 2.9.4 should have been a minor 2.10 because of the enhancement (not just a bug fix, but adding a feature should have made it a minor ... https://semver.org/)

So, if you think this is fixing a bug, we could make it 2.9.5, if you call this adding a feature, then we should use 2.10.

Personally I am in favor of a 2.10, especially with adding the bulk analyzing feature and the decision we made in the last meeting.

Although after looking at the Milestone, I don't think we will add all those features ...

**Pull request merged**
